cogs5e/models/sheet/player.py

```diff
@@ -89,67 +93,104 @@ class CustomCounter:
-                "display_type": self.display_type, "live_id": self.live_id}
+                "display_type": self.display_type, "live_id": self.live_id, "reset_to": self.reset_to,
+                "reset_by": self.reset_by}
 
     @classmethod
-    def new(cls, character, name, minv=None, maxv=None, reset=None, display_type=None, live_id=None):
+    def new(cls, character, name, minv=None, maxv=None, reset=None, display_type=None, live_id=None,
+            reset_to=None, reset_by=None):
         if reset not in ('short', 'long', 'none', None):
             raise InvalidArgument("Invalid reset.")
         if any(c in name for c in ".$"):
             raise InvalidArgument("Invalid character in CC name.")
-        if minv is not None and maxv is not None:
+        if display_type == 'bubble' and (maxv is None or minv is None):
+            raise InvalidArgument("Bubble display requires a max and min value.")
+
+        # sanity checks
+        if maxv is None and reset not in ('none', None):
+            raise InvalidArgument("Reset passed but no maximum passed.")
+        if reset_to is not None and reset_by is not None:
+            raise InvalidArgument("Both `resetto` and `resetby` arguments found.")
+
+        min_value = None
+        if minv is not None:
+            min_value = character.evaluate_math(minv)
+
+        max_value = None
+        if maxv is not None:
             max_value = character.evaluate_math(maxv)
-            if max_value < character.evaluate_math(minv):
+            if min_value is not None and max_value < min_value:
                 raise InvalidArgument("Max value is less than min value.")
             if max_value == 0:
                 raise InvalidArgument("Max value cannot be 0.")
-        if reset and maxv is None:
-            raise InvalidArgument("Reset passed but no maximum passed.")
-        if display_type == 'bubble' and (maxv is None or minv is None):
-            raise InvalidArgument("Bubble display requires a max and min value.")
 
-        if maxv:
-            value = character.evaluate_math(maxv)
-        else:
-            value = 0
-        return cls(character, name.strip(), value, minv, maxv, reset, display_type, live_id)
+        reset_to_value = None
+        if reset_to is not None:
+            reset_to_value = character.evaluate_math(reset_to)
+            if min_value is not None and reset_to_value < min_value:
+                raise InvalidArgument("Reset to value is less than min value.")
+            if max_value is not None and reset_to_value > max_value:
+                raise InvalidArgument("Reset to value is greater than max value.")
+
+        if reset_by is not None:
+            try:
+                d20.parse(reset_by)
+            except d20.RollSyntaxError:
+                raise InvalidArgument(f"{reset_by} (`resetby`) cannot be interpreted as a number or dice string.")
+
+        # set initial value
+        initial_value = max(0, min_value or 0)
```

Comment:
Would this mean I can't have an initial value less then 0?

Reply:
No - if `resetto` is specified it will be the initial value. This line of code simply sets the default if neither `resetto` or `max` is given (in which case, it is `0` or the min, whichever is larger).

Reply:
Ah, gotcha. Didn't read it as thoroughly as I thought :D